src/main/java/Task.java

```diff
@@ -0,0 +1,49 @@
+import java.time.LocalDateTime;
+import java.util.*;
+public class Task {
+    private final String val;
+    private final boolean done;
+
+    public Task(String input, String dummy) {
+        this.val = input;
+        done = false;
+    }
+
+    public Task(String input, boolean done, String dummy) {
+        this.val = input;
+        this.done = done;
+    }
+
+    public Task markDone() {
+        return new Task(this.val, true, "");
+    }
+
+    public Task markUndone() {
+        return new Task(this.val, false, "");
+    }
+
+    public boolean getDone() {
```

Comment:
Consider named the boolean method to sound like boolean, for example, you can change it to isDone.